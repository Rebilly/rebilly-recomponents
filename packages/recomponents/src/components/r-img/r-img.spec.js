import {shallowMount} from '@vue/test-utils';
import {renderToString} from '@vue/server-test-utils';
import RImg from './r-img.vue';

describe('r-img.vue', () => {
    it('renders props.msg when passed', () => {
        const wrapper = shallowMount(RImg, {
            propsData: {
                src: 'https://www.rebilly.com/wp-content/uploads/2019/01/new_features@2x-319x150.png',
            },
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('should render via SSR and match snapshot', async () => {
        const wrapper = renderToString(RImg, {
            propsData: {
                src: 'https://www.rebilly.com/wp-content/uploads/2019/01/new_features@2x-319x150.png',
            },
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('should trigger load and error events', async () => {
        const wrapper = shallowMount(RImg, {
            propsData: {
                src: 'https://www.rebilly.com/wp-content/uploads/2019/01/new_features@2x-319x150.png',
            },
        });

        wrapper.vm.onload();
        wrapper.vm.onerror();

        expect(wrapper.emitted().onload).toBeTruthy();
        expect(wrapper.emitted().error).toBeTruthy();
    });

    it('shouldn render data-url attribute if lazy is true', async () => {
        const src = 'https://www.rebilly.com/wp-content/uploads/2019/01/new_features@2x-319x150.png';
        const wrapper = shallowMount(RImg, {
            propsData: {
                src,
                lazy: true,
            },
        });
        expect(wrapper.find('img').attributes('data-url')).toBe(src);
    });
});
