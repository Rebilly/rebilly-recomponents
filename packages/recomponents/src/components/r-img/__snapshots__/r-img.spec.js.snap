// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`r-img.vue renders props.msg when passed 1`] = `
<figure role="img" class="r-component r-img"><img src="https://www.rebilly.com/wp-content/uploads/2019/01/new_features@2x-319x150.png" alt="Image" class="r-img-image">
  <div class="r-img-loading-spinner"></div>
</figure>
`;

exports[`r-img.vue should render via SSR and match snapshot 1`] = `
<figure role="img" class="r-component r-img"><img src="https://www.rebilly.com/wp-content/uploads/2019/01/new_features@2x-319x150.png" alt="Image" class="r-img-image">
  <div class="r-img-loading-spinner"></div>
</figure>
`;
